lib/hanami/validations/form.rb

```diff
@@ -12,57 +10,65 @@ module Validations
     # @since 0.6.0
     #
     # @example
-    #   require 'hanami/validations/form'
+    #   require "hanami/validations/form"
     #
     #   class Signup
     #     include Hanami::Validations::Form
     #
     #     validations do
-    #       required(:name).filled(:str?)
-    #       optional(:location).filled(:str?)
+    #       required(:name).value(:string)
+    #       optional(:location).value(:string)
     #     end
     #   end
     #
-    #   result = Signup.new('location' => 'Rome').validate
+    #   result = Signup.new("location" => "Rome").validate
     #   result.success? # => false
     #
-    #   result = Signup.new('name' => 'Luca').validate
+    #   result = Signup.new("name" => "Luca").validate
     #   result.success? # => true
     #
     #   # it works with symbol keys too
-    #   result = Signup.new(location: 'Rome').validate
+    #   result = Signup.new(location: "Rome").validate
     #   result.success? # => false
     #
-    #   result = Signup.new(name: 'Luca').validate
+    #   result = Signup.new(name: "Luca").validate
     #   result.success? # => true
     #
-    #   result = Signup.new(name: 'Luca', location: 'Rome').validate
+    #   result = Signup.new(name: "Luca", location: "Rome").validate
     #   result.success? # => true
     module Form
+      # @since 2.0.0
+      # @api private
+      class BaseValidator < Dry::Validation::Contract
+        params do
+          optional(:_csrf_token).filled(:str?)
```

Comment:
is it possible to make this attribute custom or do we need it with this name?

Reply:
This should be `optional(:_csrf_token).filled(:string)` - it will work the same, but it'll include the type spec.

Reply:
@AlfonsoUceda This is for backward compatibility with Hanami 1.x. See https://github.com/hanami/controller/blob/master/lib/hanami/action/params.rb#L119

Reply:
@solnic I changed it with: https://github.com/hanami/validations/pull/180/commits/02596c30962ac63a7ce67da15207ba650f6e8b25